plugins/com.google.cloud.tools.eclipse.sdk.ui/src/com/google/cloud/tools/eclipse/sdk/ui/preferences/CloudSdkPreferenceArea.java

```diff
@@ -148,11 +149,14 @@ private Path getDefaultSdkLocation() {
     }
   }
 
-  protected boolean validateSdk(Path location) {
+  private boolean validateSdk(Path location) {
     try {
       CloudSdk sdk = new CloudSdk.Builder().sdkPath(location).build();
       sdk.validateCloudSdk();
       sdk.validateAppEngineJavaComponents(); 
+    } catch (AppEngineJavaComponentsNotInstalledException ex) {
+      status = new Status(IStatus.WARNING, getClass().getName(),
+          MessageFormat.format(SdkUiMessages.AppEngineJavaComponentsNotInstalled, ex.getMessage()));
```

Comment:
An alternative here might be to have a hidden `Label` that we expose in this case that contains instructions to do an update.

Or perhaps have a `Link` to trigger the update on their behalf?

> Your Cloud SDK installation is missing required App Engine Java components.  &lt;a&gt;Click to install&lt;/a&gt;.